### PR TITLE
14426 Duplicate Secured Party modal instead of error message under 'Business Legal Name'

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity#14426

*Description of changes:*
- Moved duplicate secured party check after validation check so that invalid parties cannot be compared.
- Fixed bug Divya found in 14175 regarding amendments being blocked. Made changes to IsObjectEqual to handle undefined inputs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/utils/validation-helper.ts [hunk 1]
```diff
@@ -347,6 +347,9 @@ export function isObjectEqual (object1: any, object2: any): boolean {
   //
   // Does an equality between 2 objects
   //
+  if (object1 === undefined) object1 = null
+  if (object2 === undefined) object2 = null
+  // these ifs as Object.create() needs null or defined object
```
Comment (ChasBoucher) [inline ppr-ui/src/utils/validation-helper.ts]: Looks good..

**Review by ChasBoucher — APPROVED** [review]
Looks Good

File: ppr-ui/src/utils/validation-helper.ts (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/utils/validation-helper.ts]: Not really. There are other cleaner ways to do it. We can either have default values as arguments or when creating an Object: `isObjectEqual (object1: any = null, object2: any = null)` or `Object.create(object1 || null)`

I'm just unsure if we even need to check if objects are equal if one is undefined/null. We can simply do `if (!object1 || !object2) return false`.

**Review by dimak1 — CHANGES_REQUESTED** [review]

**Comment by owillborn** [comment]
@dimak1 I think you're right about returning false if either is undefined. Only cases that one of the objects could be undefined is when comparing a business to an individual, or there are no parties in the table. Both of these cases have no outcomes where a duplicate should be detected, I'll make that change. Thank you!

**Comment by owillborn** [comment]
> @dimak1 I think you're right about returning false if either is undefined. Only cases that one of the objects could be undefined is when comparing a business to an individual, or there are no parties in the table. Both of these cases have no outcomes where a duplicate should be detected, I'll make that change. Thank you!

@dimak1 looking on it now, since it is a util function and some comparisons may be made with undefined values in the future, I'm gonna go with the default values. More of a templated approach imo

**Review by cameron-eyds — APPROVED** [review]
Good clean up, thank you 👍 
Default values are much cleaner.

**Comment by dimak1** [comment]
@owillborn I see what you are saying... that there could be a case when we need to compare both undefined objects? Hmm ok then. Thanks for the clarification.

**Review by dimak1 — APPROVED** [review]

**Review by doug-lovett — APPROVED** [review]

**Pull request merged**